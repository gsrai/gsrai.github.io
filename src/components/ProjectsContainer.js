import React from 'react'
import ProjectTile from './ProjectTile'
import PropTypes from 'prop-types'

export default class ProjectsContainer extends React.Component {
  createProjectsView = (model) => {
    return model.projects.map((project, id) => {
      return (
        <ProjectTile
          key={id}
          title={project.title}
          description={project.desc}
          links={project.links} imgsrc={project.imgsrc} />
      )
    })
  }

  render() {
    return (
      <div>
        <main className={'main-content'}>
          {this.createProjectsView(this.props.projectsModel)}
        </main>
      </div>
    )
  }
}

ProjectsContainer.propTypes = {
  projectsModel: PropTypes.object.isRequired
}
