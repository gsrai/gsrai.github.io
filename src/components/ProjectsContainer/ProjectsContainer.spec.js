import { shallow } from 'enzyme'
import React from 'react'
import ProjectsContainer from './index'

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const props = {
      projectsModel: {
        projects: [{
          title: 'Some Totally Awesome Project',
          description: 'I did some stuff.',
          imgsrc: 'assets/foo.jpeg',
          links: [{ href: 'some/url', title: 'Github' }]
        }]
      }
    }
    const wrapper = shallow(<ProjectsContainer {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
