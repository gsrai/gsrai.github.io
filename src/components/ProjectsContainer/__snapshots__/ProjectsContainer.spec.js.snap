// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<main
  className="main-content"
>
  <ProjectTile
    imgsrc="assets/foo.jpeg"
    key="0"
    links={
      Array [
        Object {
          "href": "some/url",
          "title": "Github",
        },
      ]
    }
    title="Some Totally Awesome Project"
  />
</main>
`;
